### PR TITLE
Implementing IsConnected

### PR DESCRIPTION
**Pull request by salarcode**

### CODE REVIEW

File: TLSharp.Core/TelegramClient.cs [hunk 1]
```diff
@@ -377,8 +377,15 @@ private void OnUserAuthenticated(TLUser TLUser)
             _session.SessionExpires = int.MaxValue;
 
             _session.Save();
-        }
-
+        }
+
+        public bool IsConnected()
```
Comment (knocte) [inline TLSharp.Core/TelegramClient.cs]: Why do you make this a method and the other a property? I was expecting some consistency here :)

**Comment by salarcode** [comment]
Don't know why implementing connection KeepAlive commit got it in the way. Any was KeepAlive implementation is on the way.

File: TLSharp.Core/Network/TcpTransport.cs [hunk 2]
```diff
@@ -12,7 +12,7 @@ public class TcpTransport : IDisposable
         private readonly TcpClient _tcpClient;
         private int sendCounter = 0;
 
-        public TcpTransport(string address, int port, TcpClientConnectionHandler handler = null)
+        public TcpTransport(string address, int port, bool keepAlive = false, TcpClientConnectionHandler handler = null)
```
Comment (knocte) [inline TLSharp.Core/Network/TcpTransport.cs]: Can you separate this commit into its own PR please?

Comment (salarcode) [inline TLSharp.Core/Network/TcpTransport.cs]: Github did it automatically. How to fix it?

File: TLSharp.Core/TelegramClient.cs (hunk 1 above)
Comment (salarcode) [inline TLSharp.Core/TelegramClient.cs]: Because it is a resource intensive inquiry that needed to be a method.

File: TLSharp.Core/Network/TcpTransport.cs (hunk 2 above)
Comment (knocte) [inline TLSharp.Core/Network/TcpTransport.cs]: github cannot push code from you by mistake :) you pushed it to the wrong branch ;)

File: TLSharp.Core/TelegramClient.cs (hunk 1 above)
Comment (knocte) [inline TLSharp.Core/TelegramClient.cs]: Resource demanding? It's only checking for null and then later returning a property.

Comment (salarcode) [inline TLSharp.Core/TelegramClient.cs]: Yes, resource demanding underneath TcpClient

**Comment by salarcode** [comment]
I have reverted KeepAlive commit for you to have a clean merge.

File: TLSharp.Core/TelegramClient.cs (hunk 1 above)
Comment (salarcode) [inline TLSharp.Core/TelegramClient.cs]: Also there is more change coming to this in next commit.

Comment (knocte) [inline TLSharp.Core/TelegramClient.cs]: What do you mean? The method as it is now is not resource demanding so please make it a property. If in a second PR it becomes resource demanding, then convert it to a method in the next PR ;)

Comment (salarcode) [inline TLSharp.Core/TelegramClient.cs]: Ok then. dealing with your policy of clean commit takes time. Maybe another time.

Comment (knocte) [inline TLSharp.Core/TelegramClient.cs]: @salarcode I've fixed it myself and merged manually, feel free to propose the KeepAlive PR now ;)
